tests/_internal/test_find_files.py

```diff
@@ -2,118 +2,167 @@
 # Copyright (C) 2020 Arm Mbed. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 #
-from unittest import TestCase, mock
+import contextlib
 from pathlib import Path
-from pyfakefs.fake_filesystem_unittest import patchfs
+from tempfile import TemporaryDirectory
+from unittest import TestCase, mock
+from typing import Iterable
+
 from mbed_targets import MbedTargetBuildAttributes
 
-from mbed_build._internal.find_files import (
-    find_files,
-    exclude_using_mbedignore,
-    exclude_using_target_labels,
-    exclude_using_labels,
-)
-
-
-class TestFindFiles(TestCase):
-    @patchfs
-    def test_finds_files_by_name_in_given_root(self, fs):
-        files = [
-            Path("root", "folder_1", "file.txt"),
-            Path("root", "folder_1", "folder_1_1", "file.txt"),
-            Path("root", "folder_2", "file.txt"),
-        ]
-        for file in files:
-            fs.create_file(file)
+from mbed_build._internal.find_files import find_files, MbedignoreFilter, LabelFilter, BoardLabelFilter
 
-        subject = find_files("file.txt", directory=Path("root"))
 
-        self.assertEqual(list(subject), files)
+@contextlib.contextmanager
+def create_files(files: Iterable[Path]):
+    with TemporaryDirectory() as temp_directory:
+        temp_directory = Path(temp_directory)
+        for file in files:
+            file_path = temp_directory / file
+            file_directory = file_path.parent
+            if not file_directory.exists():
```

Comment:
I don't think you need this check if you do `file_directory.mkdir(parents=True, exist_ok=True)` https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir.

Reply:
You're right. I was keen to save some time by doing this, but the time difference is negligible. 👍

Reply:
https://github.com/ARMmbed/mbed-build/pull/10/commits/ea79bce0cf7c3eccef7cac4d92fa45bb9f164351

Reply:
Huh, seems to be causing a test to fail. Maybe they aren't the same 🤔

Reply:
Ah I think its `exist_ok` and not `exists_ok`.

Reply:
Yup, fixed now.